tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -209,7 +209,9 @@ private static void TryAddEndPoint(Span span)
         private void Report(ITransport transport, Span span, WafMatch[] results, bool blocked)
         {
             span.SetTag(Tags.AppSecEvent, "true");
-            span.SetTraceSamplingPriority(SamplingPriority.UserKeep);
+            var samplingPirority = _settings.KeepTraces
+                ? SamplingPriority.UserKeep : SamplingPriority.AutoReject;
+            span.SetTraceSamplingPriority(samplingPirority);
```

Comment:
@robertpi Isn't this changing the sampling priority when `KeepTraces==false`? The sampling priority could be `AutoReject` or `AutoKeep`, based on the sampling decision (or the user may have specified something else). Seems like we shouldn't be changing it in this case?